apps/system/js/statusbar.js

```diff
@@ -14,597 +14,567 @@
+          /* falls through */
+        case 'apptitlestatechanged':
+        case 'activitytitlestatechanged':
+          this.setAppearance();
+          if (!this.isPaused()) {
+            this.element.classList.remove('hidden');
+          }
+          break;
+        case 'homescreenopened':
+          // In some cases, if the user has been switching apps so fast and
+          // quickly he press the home button, we might miss the
+          // |sheets-gesture-end| event so we must resume the statusbar
+          // if needed
+          this.setAppearance();
+          if (this._pausedForGesture) {
+            this.resumeUpdate();
+            this._pausedForGesture = false;
+          }
+          this.element.classList.remove('hidden');
+          this.element.classList.remove('fullscreen');
+          this.element.classList.remove('fullscreen-layout');
+          break;
+        case 'activitydestroyed':
+          this._updateMinimizedStatusbarWidth();
+          break;
+        case 'updatepromptshown':
+          this.element.classList.remove('light');
+          break;
+        case 'updateprompthidden':
+          this.setAppearance();
+          break;
+      }
+    },
+
+    setAppearance: function() {
+      var app = Service.query('getTopMostWindow');
+      if (!app) {
+        return;
+      }
 
-    // Fetch top-most window to figure out color theming.
-    var topWindow = app.getTopMostWindow();
-    if (topWindow) {
       this.element.classList.toggle('light',
-        !!(topWindow.appChrome && topWindow.appChrome.useLightTheming())
+        !!(app.appChrome && app.appChrome.useLightTheming())
       );
 
       this.element.classList.toggle('fullscreen',
-        topWindow.isFullScreen()
+        app.isFullScreen()
       );
 
       this.element.classList.toggle('fullscreen-layout',
-        topWindow.isFullScreenLayout()
+        app.isFullScreenLayout()
       );
-    }
 
-    this.element.classList.toggle('maximized',
-      app.isHomescreen || app.isAttentionWindow ||
-      !!(app.appChrome && app.appChrome.isMaximized()));
-  },
-
-  _getMaximizedStatusBarWidth: function sb_getMaximizedStatusBarWidth() {
-    // Let's consider the style of the status bar:
-    // * padding: 0 0.3rem;
-    return Math.round((window.layoutManager ?
-      layoutManager.width : window.innerWidth) - (3 * 2));
-  },
-
-  _updateMinimizedStatusBarWidth: function sb_updateMinimizedStatusBarWidth() {
-    var app = Service.currentApp;
-    app = app && app.getTopMostWindow();
-    var appChrome = app && app.appChrome;
-
-    // Only calculate the search input width when the chrome is minimized
-    // Bug 1118025 for more info
-    if (appChrome && appChrome.isMaximized()) {
-      this._updateIconVisibility();
-      return;
-    }
+      this.element.classList.toggle('maximized', app.isHomescreen ||
+        !!(app.appChrome && app.appChrome.isMaximized()) ||
+           app.isLockscreen);
```

Comment:
We removed the check for `app.isAttentionWindow` that we need for bug 1159885. Also, where does the `app.isLockscreen` part come from?